src/token.ts

```diff
@@ -0,0 +1,29 @@
+import { Document } from "@govtechsg/open-attestation";
+import { getIssuer } from "./util/token";
+import { TokenRegistry } from "./registry";
+
+/**
+ * Class Token to interact with ERC721 contract.
+ */
+export class Token {
+  document: Document;
+
+  web3Provider: any;
+
+  tokenRegistry: TokenRegistry;
+
+  constructor(document: Document, web3Provider?: any) {
+    this.document = document;
+    this.web3Provider = web3Provider;
+    this.tokenRegistry = new TokenRegistry({ contractAddress: getIssuer(document).tokenRegistry, web3Provider });
```

Comment:
`web3Provider` is not an optional parameter in `TokenRegistry` but it's here, is it expected ?

Reply:
It looks expected according to the TODO below, but I'm surprised TS doesnt complain :grimacing: